### ISSUE TITLE
Nested concurrency

### ISSUE THREAD

**Issue by aliabbasjp** [alex-sherman/deco#46]
Is it possible to have a  @synchronised  function whose body calls another function which is decorated with @synchronised decorator?

**Comment by alex-sherman** [comment]
Yeah this should definitely work. The error you're getting is from a bug in the latest release. I've just pushed the fix to pip, so if you `pip install deco --upgrade` everything should work fine.

**Comment by JackKenney** [comment]
Similarly, is there a way to have @concurrent functions act like @synchronized functions and call @concurrent functions?
In a triple-nested for loop situation (e.g. comparing L samples against M data points for N features)
I understand if it's a limitation of the parallel pooling, but if it's not that would be great.

**Comment by alex-sherman** [comment]
I made a change to allow callables besides just functions as arguments to `@concurrent`. I tried the following:
```python
from deco import *
import time

@concurrent
def inner(i):
    time.sleep(0.1)
    return i

@concurrent
@synchronized
def middle(j):
    results = {}
    for i in range(10):
        results[i] = inner(i * j)
    return sum([v for _, v in results.items()])

@synchronized
def outer():
    results = {}
    for j in range(10):
        results[j] = middle(j)
    return sum([v for _, v in results.items()])


if __name__ == "__main__":
    start = time.time()
    print(outer())
    print(time.time() - start)
```
Which produced the following multiprocessing error:

`AssertionError: daemonic processes are not allowed to have children`

I think I agree in the abstract that it's common for child _executions_ to create more _executions_ but I think it's generally not supported by spawning additionally processes.

There is probably attempted concurrent calls in child processes to be instead communicated back to the main process and executed in the same pool, but it seems pretty complicated. I think deco is good only at making simple concurrency tasks very easy, and nested concurrency IMO beings to leave the "simple" category anyway where you may just have to bite the bullet and use multiprocessing.pool directly.

I'll leave this open as a blue sky feature, but can't commit any time to it in the near term.

**Comment by JackKenney** [comment]
That all makes sense - I agree it starts to become exceedingly complicated.
FWIW, in multiprocessing you can start child processes that start mapped pools, but pools can't start other pools they have to be fully-fledged processes to do that and at that point the OS will start to complain. 
Don't think it's worth a ton of time exploring tbh. But thanks for looking into it!
